src/Model/Casts/DateTime/BaseDateTimeCast.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Mongolid\Model\Casts\DateTime;
+
+use DateTimeInterface;
+use MongoDB\BSON\UTCDateTime;
+use MongoDB\BSON\UTCDateTimeInterface;
+use Mongolid\Model\Casts\CasterInterface;
+use Mongolid\Model\Casts\CastInterface;
+
+abstract class BaseDateTimeCast implements CastInterface
+{
+    /**
+     * @param UTCDateTime|null $value
+     * @return DateTimeInterface|null
+     */
+    abstract public function get(mixed $value): DateTimeInterface|null;
+
+    /**
+     * @param DateTimeInterface|UTCDateTimeInterface|null $value
+     * @return UTCDateTime|null
+     */
+    public function set(mixed $value): UTCDateTime|null
```

Comment:
This method doesn't look like a set but a get

Reply:
Depois veja esse thread https://github.com/leroy-merlin-br/mongolid/pull/220#discussion_r1360663535

Reply:
Olhei lá, mas mesmo assim ainda acho que esse método não se comporta como um `set` e sim como um `get`

Reply:
O @gabrielgomes94 gostou do get/set, e pra mim como o projeto é muito inspirado no Eloquent, acho que faz sentido usar a mesma nomenclatura que eles usam lá.

Reply:
@williancs Dá uma olhada: https://github.com/leroy-merlin-br/mongolid/pull/220#discussion_r1365895061

Embora esse método se chame `set`, ele não é um setter clássico  da POO. 

Ele é só um conversor de dados e tem esse nome, como o @edipoReboucas falou, inspirado na implementação do Eloquent e o seu papel aqui é só de servir de interface pra dizer que o caster vai escrever o valor castado no banco. 

Inclusive, da mesma forma que nomeamos `get/set`, também poderia ser `read/write`  ou algo do gênero.